### ISSUE TITLE
Latest make command on master needs additional fixes

### ISSUE THREAD

**Issue by astral-matrix** [shoggoth-systems/shoggoth#14]
Ok, I was able to get the make command to run but needed to fix these issues below (let me know if you want me to open another issue to track):

1. the make command failed with error:
ld: warning: -L path '/usr/local/lib' is not a directory
ld: library not found for -lnetlibc
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [link-objects-flat] Error 1

This is because make had to download and install libnetlibc (because it isnt a default package on macOS) and put it inside "/usr/local/lib" However the /usr/local/lib directory may not exist because macOS doesn't put anything under /usr/local anymore. The "/usr/local/include commands in the make worked because the make file has  mkdir -p /usr/local/include/* commands which created the directory. 
So to fix this issue in the make script, there needs to be a check for "/usr/local/lib" and if it doesn't exist it should be created before libnetlibc is installed

2. After fixing above, I got the error "/bin/sh: md5sum: command not found" because md5sum is not a standard package and the make command did not have logic to download and install it if needed.

To fix the md5sum error I installed it via homebrew as part of the coreutils (this is preferable than brew install md5sum because it also installs other check sum commands):
brew install coreutils

After fixing the 2 issues above the make and make install were successful. Thank you!

**Comment by astral-matrix** [comment]
Sorry there was no branch tag to reference but I pulled from latest on "/thenetrunna/shoggoth/tree/master" with this latest commit:
https://github.com/thenetrunna/shoggoth/commit/5f760c6241c739d21dc64beb4b77e8e55689905c

Also this relates to https://github.com/thenetrunna/shoggoth/pull/1

**Comment by thenetrunna** [comment]
Fixed by #15 

Pleease fetch and build again. Thanks!

**Comment by samsararider** [comment]
@thenetrunna thought to mention - I also had an issue with netlibc (I didn't report it though). #15 fixes the netlibc issue for me.

**Comment by thenetrunna** [comment]
> @thenetrunna thought to mention - I also had an issue with netlibc (I didn't report it though). #15 fixes the netlibc issue for me.

Great!!

**Comment by astral-matrix** [comment]
Yes latest master built fine but I did get a copy error at the end (this also occurred before but I missed catching it):
cp: /Users/dad/bin/shog: Permission denied
Installation complete!

**Comment by samsararider** [comment]
@astral-matrix you can run the binary out of `target` folder:
```
cd target
./shog-flat
```

**Comment by thenetrunna** [comment]
> Yes latest master built fine but I did get a copy error at the end (this also occurred before but I missed catching it): cp: /Users/dad/bin/shog: Permission denied Installation complete!

Did this issue occur on `make` or `make install`?

**Comment by astral-matrix** [comment]
The error appeared at the end of the make install. I am able to use the shog command using it's path $HOME/shoggoth/bin/shog but it's not in my command path. Was make supposed to copy it to /usr/local/bin or elsewhere?

**Comment by thenetrunna** [comment]
it was supposed to copy a shell script `./scripts/shog.sh` to `$HOME/bin` 

`./scripts/shog.sh` simply invokes `$HOME/shoggoth/bin/shog` and passes all its arguments to it.

**Comment by thenetrunna** [comment]
Shoggoth is not installed globally, but only for the current user, thus it is not copied to `/usr/local/bin`

It should not need `sudo` to install. It seems like the issue you have is due to `cp ./scripts/shog.sh ~/bin/shog` not working because it is not running with `sudo`

**Comment by astral-matrix** [comment]
I see what happened. The install.sh script added the ~/bin/shog path to the .bashrc in my home directory but I'm running Z shell (zsh). There was no .zshrc file in my $HOME directory.  When I switch to bash shell it has the shog command in my path.

**Comment by thenetrunna** [comment]
Oh! interesting

**Comment by thenetrunna** [comment]
why was there no .zshrc in $HOME? does zsh allow that?

**Comment by astral-matrix** [comment]
macos reads ~/.profile to for z-shell variables. I manually added 
export PATH=$PATH:~/bin
to the ~/.profile and then I was able to see it in my z-shell $PATH

A couple other things- 

Every time I run the make install command, the "export PATH=$PATH:~/bin" is appended to the .bashrc file which causes duplicate entries to the $PATH variable. 
It will be hairy to try to prevent this duplication, especially if the install script is updated to write to the .profile file, because there are other entries and possibly other $PATH entries in the file.

***Re: SUDO issue earlier, that was on my end, I previously ran sudo make/ sudo make install while testing and it created the ~/bin and ~/shoggoth directories as root owner.  I deleted them and re-ran make and make install and there were no errors.
